packages/xo-web/src/xo-app/hub/resource.js

```diff
@@ -0,0 +1,257 @@
+              <Icon icon='add-vm' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+
+        setHubInstallLoadingState({
+          ...hubInstallLoadingState,
+          [id]: true,
+        })
+        for (const pool of resourceParams.pools) {
+          try {
+            await downloadAndInstallResource({
+              namespace,
+              id,
+              version,
+              sr: pool.default_SR,
+            })
+            success('XVA import', _('hubSuccessfulInstallMsg'))
+          } catch (_error) {
+            error('Error', _error.message)
+          }
+        }
+        setHubInstallLoadingState({
+          ...hubInstallLoadingState,
+          [id]: false,
+        })
+      },
+      async create() {
+        const { isFromSources, isPoolCreated, installedTemplates } = this.state
+        const { name } = this.props
+        if (isFromSources) {
+          subscribeAlert()
+          return
+        }
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm poolPredicate={isPoolCreated} {...props} />
+          ),
+          header: (
+            <span>
+              <Icon icon='add-vm' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+        const { $pool } = resourceParams.pool
+        const { id } = find(installedTemplates, { $pool })
+        this.props.router.push(`/vms/new?pool=${$pool}&template=${id}`)
+      },
+      async deleteTemplates(__, { name }) {
+        const { isPoolCreated } = this.state
+        const resourceParams = await form({
+          render: props => (
+            <ResourceForm
+              delete
+              multi
+              poolPredicate={isPoolCreated}
+              {...props}
+            />
+          ),
+          header: (
+            <span>
+              <Icon icon='vm-delete' /> {name}
+            </span>
+          ),
+          size: 'medium',
+        })
+        const _templates = filter(this.state.installedTemplates, template =>
+          find(resourceParams.pools, { $pool: template.$pool })
+        )
+        await deleteTemplates(_templates)
+      },
+      updateSelectedInstallPools(_, selectedInstallPools) {
+        return {
+          selectedInstallPools,
+        }
+      },
+      updateSelectedCreatePool(_, selectedCreatePool) {
+        return {
+          selectedCreatePool,
+        }
+      },
+      redirectToTaskPage() {
+        this.props.router.push('/tasks')
+      },
+    },
+    computed: {
+      isFromSources: () => getXoaPlan(process.env.XOA_PLAN) === 'Community',
+      installedTemplates: (_, { id, templates }) =>
+        filter(templates, ['other.xva_id', id]),
+      isTemplateInstalledOnAllPools: ({ installedTemplates }, { pools }) =>
+        installedTemplates.length > 0 &&
+        installedTemplates.every(pool =>
+          installedTemplates.some(template => template.$pool === pool.id)
```

Comment:
You can use `find` here as well.

Reply:
I think that `some` fits well the case here, so please let it as it is.